src/sql/names.rs

```diff
@@ -32,20 +32,25 @@ impl fmt::Display for FullName {
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub enum DatabaseSpecifier {
     Ambient,
+    Temporary,
     Name(String),
 }
 
 impl fmt::Display for DatabaseSpecifier {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             DatabaseSpecifier::Ambient => f.write_str("<none>"),
+            DatabaseSpecifier::Temporary => f.write_str("mz_temp"),
             DatabaseSpecifier::Name(name) => f.write_str(name),
         }
     }
 }
 
-impl From<Option<String>> for DatabaseSpecifier {
-    fn from(s: Option<String>) -> DatabaseSpecifier {
+impl DatabaseSpecifier {
+    // Since this function is only used to load items from the Catalog database,
+    // nothing has to map to DatabaseSpecifier::Temporary.
+    // (temporary items are not stored in the database)
+    pub fn ambient_or_named(s: Option<String>) -> DatabaseSpecifier {
```

Comment:
Having this be called `from` felt misleading! I wanted the limitations of the function to be more obvious.